bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/FreeboxOsHandler.java

```diff
@@ -0,0 +1,141 @@
+import org.openhab.binding.freeboxos.internal.api.rest.FreeboxOsSession;
+import org.openhab.binding.freeboxos.internal.api.rest.RestManager;
+import org.openhab.binding.freeboxos.internal.config.FreeboxOsConfiguration;
+import org.openhab.binding.freeboxos.internal.discovery.FreeboxOsDiscoveryService;
+import org.openhab.core.audio.AudioHTTPServer;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FreeboxOsHandler} handle common parts of Freebox bridges.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class FreeboxOsHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(FreeboxOsHandler.class);
+    private final FreeboxOsSession session;
+    private final String callbackURL;
+    private final BundleContext bundleContext;
+    private final AudioHTTPServer audioHTTPServer;
+
+    private Optional<Future<?>> openConnectionJob = Optional.empty();
+
+    public FreeboxOsHandler(Bridge thing, FreeboxOsSession session, String callbackURL, BundleContext bundleContext,
+            AudioHTTPServer audioHTTPServer) {
+        super(thing);
+        this.session = session;
+        this.callbackURL = callbackURL;
+        this.bundleContext = bundleContext;
+        this.audioHTTPServer = audioHTTPServer;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Freebox OS API handler for thing {}.", getThing().getUID());
+        freeConnectionJob();
+
+        openConnectionJob = Optional.of(scheduler.submit(() -> {
+            try {
+                FreeboxOsConfiguration config = getConfiguration();
+
+                if (config.appToken.isBlank()) {
+                    updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_PENDING,
+                            "@text/info-conf-pending");
+                } else {
+                    updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE);
+                }
+
+                String currentAppToken = session.initialize(config);
+
+                if (!currentAppToken.equals(config.appToken)) {
+                    Configuration thingConfig = editConfiguration();
+                    thingConfig.put(FreeboxOsConfiguration.APP_TOKEN, currentAppToken);
+                    updateConfiguration(thingConfig);
+                    logger.info(
+                            "App token updated in Configuration, please give needed permissions to openHAB app in your Freebox management console");
```

Comment:
Shouldn't this be set as the Thing status detail message to be visible in the UI?

Reply:
I can't make this a permanent status info. It's a reminder sent to the user so he does not fall in errors due to missing rights. In that case, he'll have errors raised when using parts of the binding mentioning that some rights are missing. BTW I'm going to document this in the README.